client/components/organisms/SettingsRegister/SettingsRegister.js

```diff
@@ -0,0 +1,330 @@
+  const [email, setEmail] = useState(user.email_address)
+  const [password, setPassword] = useState(user.password);
+  const [passwordMessage, setPasswordMessage] = useState('');
+  const [usernameAvailable, setUsernameAvailable] = useState(false);
+  const [passwordValid, setPasswordValid] = useState(false);
+  const [mycity, setMyCity] = useState(user.city);
+  const [forest, setForest] = useState(user.forest_id);
+  const [acceptedTerms, setAcceptedTerms] = React.useState(user.get_update);
+ 
+
+
+  const checkPassword = (newUsername, newPassword) => {
+    const { valid, message } = validatePassword(newUsername, newPassword);
+
+    setPasswordValid(valid);
+    setPasswordMessage(message);
+  };
+
+  const checkUsername = newUsername => {
+    const { valid, message } = validateUsername(newUsername);
+
+    if (valid) {
+      setUsernameMessage('Checking username...');
+      setUsernameAvailable(false);
+
+      postCheckUsername(newUsername)
+        .then(res => {
+          setUsernameAvailable(res.available);
+          setUsernameMessage(res.message);
+        })
+        .catch(R.identity);
+    } else {
+      setUsernameAvailable(valid);
+      setUsernameMessage(message);
+    }
+  };
+
+  const updateUsername = newUserName => {
+    setUsername(newUserName);
+    checkPassword(newUserName, password);
+  };
+
+  const handleUsernameChange = e => {
+    updateUsername(e.target.value);
+    checkUsername(e.target.value);
+  };
+
+  const handlePasswordChange = e => {
+    setPassword(e.target.value);
+    checkPassword(username, e.target.value);
+  };
+ 
+ 
+  const resetState = () => {
+    setUsername(user.username);
+    setPhoneNumber(user.phone_number);
+    setBirthYear(user.birth_year);
+    // setEmail(user.email_address);
+    setPassword(user.password);
+    setMyCity(user.city);
+    setForest(user.forest_id);
+    setAcceptedTerms(user.get_update);
+  };
+
+  useEffect(() => {
+    resetState();
+  }, [user.username, user.phone_number, user.birthYear, user.password, user.city, user.forest_id, user.get_update]);
+
+  const updateUserName = e => {
+      setUsername(e.target.value);
+  };
+  const updatePhoneNumber = e => {
+    setPhoneNumber(e.target.value);
+  };
+   const updateBirthYear = e => {
+  setBirthYear(e.target.value);
+  };
+  const updateEmail = e => {
+    setEmail(e.target.value);
+    };
+  const updatePassword = e => {
+      setPassword(e.target.value);
+      };
+   const updateCity = e => {
+        setMyCity(e.target.value);
+        };
+    const updateForest = e => {
+        setForest(e.target.value);
+          };
+    const updateAcceptedTerms = e => {
+        setAcceptedTerms(e.target.value);
+            };
+          
+            // const refresh = () => dispatch(attemptGetUser())
+            // .then(resetState)
+            // .catch(R.identity);
+
+         
+            const register = () => {
+              const updatedUser = {};
```

Comment:
Should be here a call to `preventDefault`, in order to prevent the form from being submitted in query params of url.
You already did this in Register, so do it here.
```
const register  = async (e) => {
// 
        e.preventDefault();
```

Reply:
done